src/overlays/actors/ovl_En_Xc/z_en_xc.c

```diff
@@ -77,479 +48,2375 @@ static ColliderCylinderInitType1 sCylinderInit = {
+            if (action == 23) {
+                animation = &gShiekPlayingHarpAnim;
+            }
+        }
+    }
+    return animation;
+}
+
+void EnXc_CalcXZAccel(EnXc* this) {
+    f32 timer = this->timer;
+    f32* speedXZ = &this->actor.speedXZ;
+
+    if (timer < 9.0f) {
+        *speedXZ = 0.0f;
+    } else {
+        if (timer < 3.0f) {
+            *speedXZ = (((kREG(2) * 0.01f) + 1.2f) / 3.0f) * (timer - 9.0f);
+        } else {
+            *speedXZ = (kREG(2) * 0.01f) + 1.2f;
+        }
+    }
+    Actor_MoveForward(&this->actor);
+}
+
+void func_80B3D644(EnXc* this) {
+    Actor_MoveForward(&this->actor);
+}
+
+void EnXc_CalcXZSpeed(EnXc* this) {
+    f32 timer = this->timer;
+    f32* speedXZ = &this->actor.speedXZ;
+
+    if (timer < 3.0f) {
+        *speedXZ = (((kREG(2) * 0.01f) + 1.2f) / 3.0f) * (3.0f - timer);
+    } else {
+        *speedXZ = 0.0f;
+    }
+    Actor_MoveForward(&this->actor);
+}
+
+void func_80B3D6F0(EnXc* this) {
+    EnXc_CalcXZAccel(this);
+}
+
+void func_80B3D710(EnXc* this) {
+    Actor_MoveForward(&this->actor);
+}
+
+void func_80B3D730(EnXc* this) {
+    EnXc_CalcXZSpeed(this);
+}
+
+void func_80B3D750(EnXc* this, GlobalContext* globalCtx) {
+    if (EnXc_MinuetCS(this, globalCtx) && EnXc_BoleroCS(this, globalCtx)) {
+        this->action = XC_ACTION_WAIT;
+    }
+}
+
+void EnXc_SetupFallFromSkyAction(EnXc* this, GlobalContext* globalCtx) {
+    s32 pad;
+    CutsceneContext* csCtx = &globalCtx->csCtx;
+    CsCmdActorAction* npcAction;
+
+    if (csCtx->state && (npcAction = csCtx->npcActions[4], npcAction && npcAction->action == 2)) {
+        s32 pad;
+        Vec3f* pos = &this->actor.world.pos;
+        SkelAnime* skelAnime = &this->skelAnime;
+        f32 frameCount = Animation_GetLastFrame(&gShiekFallingFromSkyAnim);
+
+        this->action = XC_ACTION_GRACEFUL_FALL;
+        this->drawMode = XC_DRAW_DEFAULT;
+
+        pos->x = npcAction->startPos.x;
+        pos->y = npcAction->startPos.y;
+        pos->z = npcAction->startPos.z;
+
+        func_80B3D48C(this, globalCtx);
+        func_80B3C964(this, globalCtx);
+        Animation_Change(skelAnime, &gShiekFallingFromSkyAnim, 1.0f, 0.0f, frameCount, ANIMMODE_ONCE, 0.0f);
+        func_80B3D118(globalCtx);
+    }
+}
+
+void func_80B3D8A4(EnXc* this, GlobalContext* globalCtx, s32 animFinished) {
+    if (animFinished) {
+        SkelAnime* skelAnime = &this->skelAnime;
+        f32 frameCount = Animation_GetLastFrame(&gShiekWalkingAnim);
+
+        Animation_Change(skelAnime, &gShiekWalkingAnim, 1.0f, 0.0f, frameCount, ANIMMODE_LOOP, -8.0f);
+
+        this->action = XC_ACTION_ACCEL;
+        this->timer = 0.0f;
+
+        func_80B3C9DC(this);
+        this->actor.gravity = -((kREG(1) * 0.01f) + 13.0f);
+        this->actor.minVelocityY = -((kREG(1) * 0.01f) + 13.0f);
+    } else {
+        func_80B3C8CC(this, globalCtx);
+    }
+}
+
+void EnXc_SetupWalkAction(EnXc* this) {
+    f32* timer = &this->timer;
```

Comment:
what happens if you remove this temp

Reply:
Regalloc completely messed up. This one also took forever to match lol